nameresolution/metadata.go

```diff
@@ -33,3 +37,41 @@ type Metadata struct {
 	metadata.Base `json:",inline"`
 	Configuration interface{}
 }
+
+// GetHostAddress returns the host address property.
+func (m Metadata) GetHostAddress() string {
+	return m.Properties[HostAddress]
+}
+
+// GetDaprHTTPPort returns the Dapr HTTP port property.
+// If the port is invalid, returns 0.
+func (m Metadata) GetDaprHTTPPort() int {
+	p, _ := strconv.Atoi(m.Properties[DaprHTTPPort])
+	if p < 0 {
+		p = 0
+	}
+	return p
+}
+
+// GetDaprPort returns the host address property.
+func (m Metadata) GetDaprPort() int {
+	p, _ := strconv.Atoi(m.Properties[DaprPort])
```

Comment:
Shouldn’t we error in these cases and fail the unit of the component? I’d expect an internal type which containing the parsed metadata with the target types.

Reply:
The components error later because the port is 0. Also worth pointing out that this value is set by the runtime, so it shouldn't really happen that this be invalid, in practice.